src/sync/incomingSyncUtils.js

```diff
@@ -632,6 +640,27 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       numberSequence.addNumberToReuse(numberToReuse);
       break;
     }
+    case 'Pref': {
+      const { item } = record;
+      if (item === 'store_preferences') {
+        try {
+          const recordData = JSON.parse(record?.data);
+          const recordKeys = Object.keys(recordData);
+          recordKeys.forEach(key => {
+            const prefData = JSON.stringify(recordData[key] ?? {});
+            internalRecord = { id: key, data: prefData };
+            database.update(recordType, internalRecord);
+          });
+        } catch (error) {
+          // Silently ignore malformed prefs.
```

Comment:
good place for some snaggin' in a new issue I reckon?

Reply:
Nice! #2673.